TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -43,70 +44,163 @@ class ReadOnlyPropertiesChecker: NodeListener {
 }
 
 
-class FunctionCallChecker: NodeListener {
+    class FunctionCallChecker : NodeListener
+    {
 
-	public void OnNode(Node node, ParserRuleContext context, CodeModel.Program program) {
-		var statement = node as FunctionCaller;
+        public void OnNode(Node node, ParserRuleContext context, CodeModel.Program program)
+        {
+            var procedureStyleCall = node as FunctionCaller;
+            if (procedureStyleCall == null || procedureStyleCall.FunctionCall == null)
+                return;
+
+            List<FunctionDeclaration> functionDeclarations = new List<FunctionDeclaration>();
+            string message;
+
+            if (procedureStyleCall.FunctionCall.FunctionNameToken.TokenFamily == Scanner.TokenFamily.AlphanumericLiteral)
+                return; //We don't have to manage AlphanumericLiteral (exteral program) for the moment...
 
-            if (statement != null)
+            if (procedureStyleCall.FunctionDeclaration == null)
             {
-                foreach (var fun in statement.FunctionCalls)
+                //Get Funtion just by name and profile (matches on precise parameters)
+                functionDeclarations =
+                    node.SymbolTable.GetFunction(new URI(procedureStyleCall.FunctionCall.FunctionName),
+                        (procedureStyleCall.FunctionCall as ProcedureCall).AsProfile(node.SymbolTable));
+                var potentialVariables =
+                    node.SymbolTable.GetVariable(new URI(procedureStyleCall.FunctionCall.FunctionName));
+                //Check if there is more than one FunctionDeclaration
+                if (CheckFunctionAmbiguity(functionDeclarations, node).Count > 1)
+                    return; //Do not continue, the fonction is ambigous
+
+                if (functionDeclarations.Count == 0)
+                    //Get Funtion just by name and profile (matches only on parameters count)
+                    functionDeclarations =
+                        node.SymbolTable.GetFunction(new URI(procedureStyleCall.FunctionCall.FunctionName),
+                            (procedureStyleCall.FunctionCall as ProcedureCall).AsProfile(node.SymbolTable), false);
+
+                //Check if there is more than one FunctionDeclaration
+                if (CheckFunctionAmbiguity(functionDeclarations, node).Count > 1)
+                    return; //Do not continue, the fonction is ambiguous
+
+                if (functionDeclarations.Count == 0)
+                    //Get function just by name (no matches on parameters)
+                    functionDeclarations =
+                        node.SymbolTable.GetFunction(new URI(procedureStyleCall.FunctionCall.FunctionName));
+
+                //Check Variable and Function Name Ambiguity
+                if (potentialVariables.Count > 0 &&
+                    CheckFunctionVariableAmbiguity(potentialVariables, functionDeclarations, node.SymbolTable) > 0)
```

Comment:
As functionDeclarations contains only function with the same name, you can directly check if functionDeclarations.Count > 0